### ISSUE TITLE
[java] Unused Assigned Value

### ISSUE THREAD

**Issue by numeralnathan** [pmd/pmd#1836]
**Affects PMD Version:** 

6.14.0

**Rule:**

Either this is for a new Best Practices rule (e.g. UnusedAssignedValue) or this is for an enhancement to [UnusedLocalVariable](https://pmd.github.io/pmd-6.14.0/pmd_rules_java_bestpractices.html#unusedlocalvariable).

**Description:**

UnusedLocalVariable does not flag a problem if a local variable is used but the final assignment to the variable is not used.  In the sample code #1 below, the last assignment to the `list` variable is not flagged as a problem.  However, in the sample code #2 below, the last assignment is valid since it will be used on the next iteration of the loop.

When I am developing a new method, I may start developing with one algorithm in mind but as the code writing proceeds I have other ideas come up.  Sometimes this leads to situations where I assigned a value to a local variable but never use the value.  In the least, I should not assign the value to the variable.  In rare cases, the entire line should be removed.

**Code Sample #1 demonstrating the issue:**

```
      ArrayList<Object> list;

      list = new ArrayList<>();

      list.size();

      list = new ArrayList<>();
```

**Code Sample #2 demonstrating a valid use case:**

```
      ArrayList<Object> list;

      list = new ArrayList<>();

      while (true)
      {
         list.size();

         list = new ArrayList<>();
      }
```

**Running PMD through:**

PMD Eclipse Plugin

**Comment by oowekyala** [comment]
I believe this is fixed by [UnusedAssignment](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedassignment)?

**Comment by numeralnathan** [comment]
@oowekyala You are correct.  `UnusedAssignment` covers it.  I am closing this ticket.
